nixos/modules/services/backup/burpServer.nix

```diff
@@ -0,0 +1,83 @@
+{ config, lib, networking, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.burpServer;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.burpServer = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable burp.";
+      };
+
+      user = mkOption {
+        default = "burp";
+        description = "The user to run burp as";
+      };
+
+      group = mkOption {
+        default = cfg.user;
+        description = "The group to run burp as (default same as user)";
+      };
+
+      configFile = mkOption {
+        default = "/etc/burp/burp.conf";
+        description = "Path to the burp config file. You need to provide a fully working configuration file and setup all paths yourself.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.burp;
+        example = literalExample "pkgs.burp";
+        description = ''burp package to use.'';
+      };
+
+    };
+
+  };
+
+  ###### implementation
+
+  config = mkIf config.services.burpServer.enable {
+
+    users.extraUsers.burp = {
```

Comment:
I'm not sure what you mean about changing existing users, I didn't see another obvious way to define a user and ensure the account is setup as a daemon account.
How could I "guard" this user? 
Making it non-configurable could be a problem in some cases where the actual store is on an NFS server. In that case depending on the implementation it might actually be necessary to be able to configure the uid as well, but burp is not really targeted at those huge setups, so I decided to ignore this for now.
While the burp data store could easily be moved between systems, I believe reserving a static id for this user in nixos (and thus blocking it on every nixos system out there) is probably overkill, I have not heard of anyone besides me using burp on nixos.